controllers/grid/files/proof/form/ApprovedProofForm.inc.php

```diff
@@ -101,21 +102,29 @@ public function initData()
     public function execute(...$functionArgs)
     {
         parent::execute(...$functionArgs);
-        $submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /* @var $submissionFileDao SubmissionFileDAO */
         $salesType = $this->getData('salesType');
+
+        $params = [
+            'direct_sales_price' => $this->getData('price'),
```

Comment:
Have you tested this? At a glance, I suspect these should be setting names (`directSalesPrice`) rather than column names (`direct_sales_price`), but I could be wrong.

Reply:
The `EntityDAO::_update` uses the `DB::update()`, which accepts the column names.

More info at https://laravel.com/docs/7.x/queries#updates

Reply:
This will call `Repository::edit`, which calls `array_merge` on the supplied array and the existing `DataObject::_data` array. Then it calls to `DAO::update`, and from there `EntityDAO::_update`. This does eventually rely on `DB::update`, but it maps from property names to column names (see the `$primaryTableColumns` mapping in your `DAO`) first.

So I think this does need fixing, but beyond that, it also needs testing.